### ISSUE TITLE
Add a `validate` method or similar to useForm

### ISSUE THREAD

**Issue by allforabit** [react-hook-form/react-hook-form#3068]
**Is your feature request related to a problem? Please describe.**
It's unclear what the best approach is to handling in progress forms and work with the interim state of the form before calling `handleSubmit`. I'd like to be able to get some of the form state into a state manager quite early in some cases and it's not that straightforward to do it. Some of these issues are outlined quite well in this article: https://medium.com/@angular_evan/react-hook-form-our-implementation-6f2ed346c07

**Describe the solution you'd like**
The functionality of `handleSubmit` is to run the validation functions and depending on the result call your error or success handlers. It would be nice to have a slightly different additional method that could run the validations but not update the submitted count and similar. The success/error callbacks would be used to call into your state manager, etc to save in progress state. The signature could include the field name to only run validations on that particular field. The method would be very similar to `handleSubmit` but I think it would be very convenient and help bridge the gap between react-hook-form and state management solutions.

**Describe alternatives you've considered**
The alternative I'm currently using is calling `handleSubmit` with an alternative callback than the main `onSubmit` and this seems to work quite well although it is a little bit semantically funny and I'm not sure if it will have some unintended side effects.

**Comment by bluebill1049** [comment]
Thanks for the feature request. personally, i don't think we need such a feature, but put on the board for a couple weeks anyhow, see if others have a similar requests.

**Comment by bluebill1049** [comment]
Think twice about this, I think it's probably easier to just build with `resolver`? It's probably more suitable for return array fields level errors (variation)

**Comment by allforabit** [comment]
OK I will have a play around with this. I haven't actually used the resolver yet but have been meaning to investigate it properly. Love that there's an official resolver for all those validation libraries!

The reason I opened the feature request was to try to find a way to store a kind of work in progress form that could be returned to at a later point by the user. I'm not sure if there's a suggested way of doing this? Something I have considered is lots of mini forms similar to the multistep example in the docs (which would be added to a state manager at each step). Another thing I played around with was `watch` but I found it difficult to not trigger infinite renders using this. This is what let to the idea of using field level `onBlur` handlers to call the submit handler to get the current status of the form and dispatch accordingly to the state manager. To be honest it's not a good approach as this validates the whole form when really I only want the validation of the blurred field (to decide to put it into state).

So yes I'll see if using the `resolver` approach works better!
